pkg/ipcache/metadata.go

```diff
@@ -185,11 +185,8 @@ func InjectLabels(src source.Source, updater identityUpdater, triggerer policyTr
 		// been updated with new labels.
 		var wg sync.WaitGroup
 		updater.UpdateIdentities(idsToPropagate, nil, &wg)
-		wg.Wait()
-
-		// This will take the accumulated policy map changes from the above,
-		// and realizes it into the datapath.
-		triggerer.TriggerPolicyUpdates(false, "kube-apiserver identity updated")
+		policyImplementedWG := triggerer.UpdatePolicyMaps(context.TODO(), &wg)
+		policyImplementedWG.Wait()
```

Comment:
no harm to cross check docs in public functions if there is any update required e.g. https://github.com/cilium/cilium/blob/89989608a36090944d9d6cb91bfa4b92400fc878/pkg/ipcache/metadata.go#L520-L549

Reply:
So effectively if we were going to include BPF policymap updates in the above diagram, then there should be another parallel box next to the "IPC" and another arrow next to (4) which updates the policymaps. And the only difference between before vs. after this PR is whether that step is async vs. synchronous. Probably it makes sense to revisit this point as part of #18301 as we come up with the next design iteration of the IPCache.

Reply:
makes sense to me, thanks.